### DISCUSSION TITLE
Multi-tenancy publishing

### DISCUSSION THREAD

**Discussion by benfortuna** [websight-io/starter#56] [Q&A] [answered]
Hi Websight creators,

I am encouraged by your containerized architecture for its potential to scale up to support many websites with one author instance. 

Just curious how it might work for publishing multiple spaces: would you recommend a separate nginx configuration for each published space or can a single nginx configuration support multiple tenants?

**Comment by michalcukierman** [comment, marked as answer]
Hi Ben,

We have released another Components Library based on [Bulma](https://bulma.io/) this week. You can find it [here](https://github.com/websight-io/bulma). WebSight Starter will be configured to work with two sites (one for each template), so the example will be available soon (this or next week I guess).

The general recommendation, for now, is to use a single Nginx container with per-site virtual hosts, like here:
https://www.keycdn.com/support/nginx-virtual-host . Nginx is fast, and you can scale both horizontally and vertically in our architecture, so it's mostly about Developer Experience and code maintainability. Single configuration per `server block` allows you to organize the code well and separate site-specific logic.

Thanks to the fact, that we use containers, we are quite flexible in composing the final stack. In more complex scenarios, we can introduce multiple Nginx servers or even CMS containers (we may have a stack for application or applications groups - i.e. corporate or public sites). We are also finishing Kubernetes Helm Chart, so it will be simpler to create advanced stacks. Again, the recommendation is to keep it simple, unless you need to address some specific business request.


One more thing to add, if you plan to configure multiple sites to work with one domain, you probably will want to set simple paths shortening to false here: https://github.com/websight-io/starter/blob/main/distribution/src/main/features/luna.json#L17
This is a way to shorten outgoing links (to skip common `/content/<space>/pages` path element from outgoing links). This should not be a case for domain-per-site cases.

**Comment by michalcukierman** [comment]
@kamil-kwarciak 
Can we add a ticket to create another documentation covering:
- working with multiple applications, sites, domains, paths shortening, links rewrites, etc?

**Comment by paweljozwicki** [comment]
As @michalcukierman mentioned we used virtual hosts to define configuration per page.
For local development purposes, we used [nip.io](https://nip.io/) to define host addresses. You can find our changes [here](https://github.com/websight-io/starter/commit/c60af4174be5e6505dc86628a4b9f5da0ff3eeb2).
